base/uk.ac.stfc.isis.ibex.ui.beamstatus/src/uk/ac/stfc/isis/ibex/ui/beamstatus/views/BeamStatusGraphView.java

```diff
@@ -0,0 +1,399 @@
+ * (e.g. hourly, daily)
+ * 
+ * @author Adrian Potter
+ */
+public abstract class BeamStatusGraphView extends DataBrowserAwareView implements ModelListener {
+
+    /** View ID registered in plugin.xml. */
+    public static final String ID = "uk.ac.stfc.isis.ibex.ui.beamstatus.views.BeamStatusGraphView"; //$NON-NLS-1$
+
+    /** Title for the Y-axis. */
+    private static final String Y_AXIS_TITLE = "Beam current (μA)";
+
+    /** Upper limit for the beam current. */
+    private static final double CURRENT_UPPER = 250.0;
+
+    /** Lower limit for the beam current. */
+    private static final double CURRENT_LOWER = 0.0;
+
+    /** TS1 beam current PV name */
+    private static final String TS1_BEAM_CURRENT_PV = "AC:TS1:BEAM:CURR";
+
+    /** TS2 beam current PV name */
+    private static final String TS2_BEAM_CURRENT_PV = "AC:TS2:BEAM:CURR";
+
+    /** Synchotron beam current PV name */
+    private static final String SYNCH_BEAM_CURRENT_PV = "AC:SYNCH:BEAM:CURR";
+
+    /** Plot. */
+    private Plot plot;
+
+    /** Model of the currently active Data Browser plot or <code>null</code>. */
+    private Model model;
+
+    /** Controller that links model and plot. */
+    private Controller controller = null;
+
+    /** {@inheritDoc} */
+    @Override
+    protected void doCreatePartControl(final Composite parent) {
+        // Arrange disposal
+        parent.addDisposeListener(new DisposeListener() {
+            @Override
+            public void widgetDisposed(DisposeEvent e) {
+
+                // Be ignorant of any change of the current model after
+                // this view is disposed.
+                if (model != null) {
+                    model.removeListener(BeamStatusGraphView.this);
+                }
+            }
+        });
+
+        final GridLayout layout = new GridLayout(2, false);
+        parent.setLayout(layout);
+
+        // =====================
+        // ======= Plot ========
+        // =====================
+
+        // The canvas has to be wrapped in a composite so that the canvas has
+        // (0,0) coordinate.
+        // This is a work around for the inconsistency between
+        // figure.getBounds() and gc.getclipping().
+        Composite composite = new Composite(parent, SWT.NONE);
+        composite.setLayoutData(new GridData(SWT.FILL, SWT.FILL, true, true, layout.numColumns, 1));
+        composite.setLayout(new FillLayout());
+
+        // Double_BUFFERED is required to force RAP to use NativeGraphicsSource.
+        // By default, it uses BufferedGraphicsSource which has problem to
+        // render it in web browser.
+        final Canvas canvas = new Canvas(composite, SWT.DOUBLE_BUFFERED);
+
+        // Create plot with basic configuration
+
+        plot = Plot.forCanvas(canvas);
+        plot.getXYGraph().setTitle(getPlotTitle());
+
+        selectPV(TS1_BEAM_CURRENT_PV);
+        selectPV(TS2_BEAM_CURRENT_PV);
+        selectPV(SYNCH_BEAM_CURRENT_PV);
+
+        setYAxisName();
```

Comment:
Would this be better if you did:
`if (model != null && model.getAxisCount() == 0) {
  setAxisProperties(model.getAxisCount() - 1)
}`

Then you need not do this check at the start of all of the methods. Otherwise add model not null check to all private methods)